crates/uv-git-types/src/lib.rs

```diff
@@ -151,6 +153,29 @@ impl From<GitUrl> for Url {
 
 impl std::fmt::Display for GitUrl {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.repository)
+        write!(f, "{}", redacted_url(&self.repository))
+    }
+}
+
+/// Return a version of the URL with redacted credentials, allowing the generic `git` username (without a password)
+/// in SSH URLs, as in, `ssh://git@github.com/...`.
+// TODO(john): This code is a duplicate of the function in `uv-auth`.
+// It's here to avoid importing `uv-auth` into this crate. Consolidate these
+// once we use a `RedactedUrl` type.
+pub fn redacted_url(url: &Url) -> Cow<'_, Url> {
+    let no_credentials = url.username().is_empty() && url.password().is_none();
+    let generic_git_username =
+        url.scheme() == "ssh" && url.username() == "git" && url.password().is_none();
+    if no_credentials || generic_git_username {
+        Cow::Borrowed(url)
+    } else {
+        let mut url = url.clone();
+        if url.password().is_some() {
+            let _ = url.set_password(Some("****"));
+        // A username on its own might be a secret token.
+        } else if url.username() != "" {
+            let _ = url.set_username("****");
```

Comment:
It's a little confusing that half of our redact methods remove the credentials entirely, and then half replace them with stars. It's also a little strange (though may not matter in practice) that we're using this version for the cache key, so like the cache key path now includes these stars?

It might be nice to use a different name for these methods that replace with stars vs. those that remove the credentials entirely (like `sensitive` vs. `redact` or `without_credentials` or something).

Or just drop the stars entirely and match the existing redact logic for consistency for now.

Reply:
It can be helpful to have the stars when debugging but we can defer this decision until after this PR (on `main` we do it in some places and not others). The follow-up PR will put this decision in one place. To move this PR forward, I've removed the masking from `redacted_url` and dropped the stars.

I've also removed all uses from `uv-auth` (including around the credentials cache). As a result, I've moved `redacted_url` out of `uv-auth` and into a new `uv-redacted` crate. This means we continue to rely on the existing mechanisms for redacting credentials there. This should help move the PR forward and is still a strict improvement.